packages/gatsby-source-wordpress/src/gatsby-node.js

```diff
@@ -616,8 +631,8 @@ function recursiveAddFields(ent, newEnt) {
 function getValidName(key) {
   let nkey = key
   const NAME_RX = /^[_a-zA-Z][_a-zA-Z0-9]*$/
-  if (!NAME_RX.test(nkey)) {
-    nkey = `_${nkey}`.replace(/-/g, `_`).replace(/:/g, `_`)
+  if (!NAME_RX.test(nkey) || restrictedNodeFields.includes(nkey)) {
+    nkey = `${conflictFieldPrefix}${nkey}`.replace(/-|__|:|\.|\s/g, `_`)
```

Comment:
so we should provide a singfle robust utility for doing this (there is one in gatsby), but this is still incomplete in terms of what graphql will allow. It needs to replace any character that isn't alphanumeric or a `_`. It also can't start with a number or two or more `_`. e.g here are some fields that would still error out here.

- "##foo"
- "1number"